pkg/controller/gc_controller_test.go

```diff
@@ -191,38 +132,50 @@ func TestGarbageCollectBackup(t *testing.T) {
 	tests := []struct {
 		name                           string
 		backup                         *api.Backup
-		deleteBackupFile               bool
 		snapshots                      sets.String
 		backupFiles                    sets.String
 		backupMetadataFiles            sets.String
 		restores                       []*api.Restore
+		nilSnapshotService             bool
+		expectedErr                    bool
 		expectedRestoreDeletes         []string
 		expectedBackupDelete           string
 		expectedSnapshots              sets.String
 		expectedObjectStorageDeletions sets.String
 	}{
 		{
```

Comment:
We may just want to convert this to a few test cases:
1. nil snapshot server, backup has snapshots, return error (ensure that no DeleteSnapshot, DeleteBackupDir, restore Delete calls are issued (handled by mock.AssertExpectations etc))
1. happy path
1. do as much as you can (some DeleteSnapshot errors & successes, DeleteBackupDir error & success, some restore Delete errors & successes) and return error(s) encountered